sdk/core/azure-core/src/strings.cpp

```diff
@@ -73,18 +73,14 @@ namespace Azure { namespace Core { namespace Details {
     auto result = std::string(src);
     for (auto i = result.begin(); i < result.end(); i++)
     {
-      *i = c_LocaleInvariantUppercaseTable[static_cast<unsigned char>(*i)];
-      if (*i >= 'A' && *i <= 'Z')
-      {
-        *i += 32;
-      }
+      *i = ToLower(static_cast<unsigned char>(*i));
     }
     return result;
   }
 
-  unsigned char ToLower(const unsigned char src) noexcept
+  unsigned char ToLower(const unsigned char symbol) noexcept
   {
-    auto result = c_LocaleInvariantUppercaseTable[src];
+    auto result = c_LocaleInvariantUppercaseTable[symbol];
     if (result >= 'A' && result <= 'Z')
     {
       result += 32;

```

Comment:
I think we should have a `c_LocaleInvariantLowercaseTable` and we just return c_LocaleInvariantLowercaseTable[symbol] in this function. We should avoid the if branches.

Reply:
the `if` clause wasn't added by Billy right?

Reply:
Without the `if`, we can make full use of instruction-level parallelism, so this function can be super-fast.

Reply:
LoL.  Not added by Billy,
He added the toUpperCase.  I didn't want to add a new table for tolower. But what you said makes sense. 
I will update this with an extra table then, then tolower and toupper will be fast

Reply:
since toUpper is not used anywhere, I will change the table to tolower and update the `LocaleInvariantCaseInsensitiveEqual`